pylint/checkers/imports.py

```diff
@@ -536,7 +549,17 @@ def leave_module(self, node: nodes.Module) -> None:
         self._imports_stack = []
         self._first_non_import_node = None
 
-    def compute_first_non_import_node(self, node):
+    def compute_first_non_import_node(
+        self,
+        node: nodes.If
+        | nodes.Expr
+        | nodes.Comprehension
+        | nodes.IfExp
+        | nodes.Assign
+        | nodes.AssignAttr
+        | nodes.TryExcept
+        | nodes.TryFinally,
```

Comment:
If this can be everything except an import node it could also be a classDef or a lot of things, right ? Maybe ``NodeNG`` would make sense ?

Reply:
See L577. I thought I would be explicit just like we normally are for `visit_` methods.

That said L622 shows that L602 is incorrect.

I'm drafting as I'd like to fix that in this PR.